blueocean-core-js/src/js/paths/rest.js

```diff
@@ -28,7 +29,7 @@ export default {
 
     run({ organization, pipeline, branch, runId }) {
         if (branch) {
-            return `${this.pipeline(organization, pipeline)}branches/${encodeURIComponent(encodeURIComponent(branch))}/runs/${runId}/`;
+            return `${this.pipeline(organization, pipeline)}branches/${encodeURIComponent(JenkinsEncode.encode(branch))}/runs/${runId}/`;
```

Comment:
ughhhhhhhhhhhh this is exactly what I was talking about: just building strings for the urls